packages/core/src/api/exporters/html/htmlConversion.test.ts

```diff
@@ -107,144 +116,304 @@ describe("Test HTML conversion", () => {
-        content: "Paragraph 2",
-        children: [
-          {
-            type: "paragraph",
-            content: "Nested Paragraph 1",
-          },
-          {
-            type: "paragraph",
-            content: "Nested Paragraph 2",
-          },
-          {
-            type: "paragraph",
-            content: "Nested Paragraph 3",
-          },
-        ],
-      },
-    ];
-
-    beforeEach(() => {
-      editor.replaceBlocks(editor.document, blocks);
-    });
-
-    it("Selection within a block's children", async () => {
-      // Selection starts and ends within the first block's children.
+  // Sets the editor selection to the given start and end positions, then
+  // exports the selected content to HTML and compares it to a snapshot.
+  function testSelection(testName: string, startPos: number, endPos: number) {
+    editor.dispatch(
+      editor._tiptapEditor.state.tr.setSelection(
+        TextSelection.create(editor._tiptapEditor.state.doc, startPos, endPos)
+      )
+    );
+    if (
+      "node" in editor._tiptapEditor.view.state.selection &&
+      (editor._tiptapEditor.view.state.selection.node as Node).type.spec
+        .group === "blockContent"
+    ) {
       editor.dispatch(
         editor._tiptapEditor.state.tr.setSelection(
-          TextSelection.create(editor._tiptapEditor.state.doc, 18, 80)
+          new NodeSelection(
+            editor._tiptapEditor.view.state.doc.resolve(
+              editor._tiptapEditor.view.state.selection.from - 1
+            )
+          )
         )
       );
+    }
 
-      const copiedFragment =
-        editor._tiptapEditor.state.selection.content().content;
+    const originalSlice = editor._tiptapEditor.view.state.selection.content();
 
-      await initializeESMDependencies();
-      const exporter = createExternalHTMLExporter(editor.pmSchema, editor);
-      const externalHTML = exporter.exportProseMirrorFragment(
-        copiedFragment,
-        {}
-      );
-      expect(externalHTML).toMatchFileSnapshot(
-        "./__snapshots_fragment_edge_cases__/" +
-          "selectionWithinBlockChildren.html"
-      );
-    });
+    // Uses default ProseMirror clipboard serialization.
+    const internalHTML: string = pmView.__serializeForClipboard(
+      editor._tiptapEditor.view,
+      originalSlice
+    ).dom.innerHTML;
 
-    it("Selection leaves a block's children", async () => {
-      // Selection starts and ends within the first block's children and ends
-      // outside, at a shallower nesting level in the second block.
-      editor.dispatch(
-        editor._tiptapEditor.state.tr.setSelection(
-          TextSelection.create(editor._tiptapEditor.state.doc, 18, 97)
-        )
-      );
+    expect(internalHTML).toMatchFileSnapshot(
+      `./__snapshots_selection_html__/${testName}.html`
+    );
 
-      const copiedFragment =
-        editor._tiptapEditor.state.selection.content().content;
+    const recreatedSlice = pmView.__parseFromClipboard(
```

Comment:
this basically tests whether `__parseFromClipboard` works as the reverse from `__serializeForClipboard` right? i.e. we're basically testing prosemirror internals.